SecretSanta/src/SecretSanta.Domain/Services/GroupService.cs

```diff
@@ -32,5 +33,62 @@ public List<Group> FetchAll()
         {
             return DbContext.Groups.ToList();
         }
+
+        public List<User> GetUsers(int groupId)
+        {
+            return DbContext.Groups
+                .Where(x => x.Id == groupId)
+                .SelectMany(x => x.GroupUsers)
+                .Select(x => x.User)
+                .ToList();
+        }
+
+        public void DeleteGroup(Group group)
+        {
+            if (@group == null) throw new ArgumentNullException(nameof(@group));
+
+            DbContext.Groups.Remove(group);
+            DbContext.SaveChanges();
+        }
+
+        public User AddUserToGroup(int groupId, User user)
+        {
+            if (user == null) throw new ArgumentNullException(nameof(user));
+            
+            if (groupId <= 0) throw new ArgumentException($"Invalid groupId. {groupId} must be greater than 0");
```

Comment:
For `ArgumentException` (and its derived classes) you want to use the constructor overload that takes in the parameter name. 
```suggestion
            if (groupId <= 0) throw new ArgumentException($"Invalid {nameof(groupId)}. {groupId} must be greater than 0", nameof(groupId));
```